hathor/websocket/protocol.py

```diff
@@ -30,23 +34,135 @@ class HathorAdminWebsocketProtocol(WebSocketServerProtocol):
         can send the data update to the clients
     """
 
+    MAX_GAP_LIMIT: int = 30
+
     def __init__(self, factory: 'HathorAdminWebsocketFactory') -> None:
         self.log = logger.new()
         self.factory = factory
         self.subscribed_to: set[str] = set()
+        self._history_streamer: HistoryStreamer | None = None
+        self._manual_address_iter: ManualAddressSequencer | None = None
         super().__init__()
 
     def onConnect(self, request):
+        """Called by the websocket protocol when the connection is opened but it is still pending handshaking."""
         self.log.info('connection opened, starting handshake...', request=request)
 
     def onOpen(self) -> None:
+        """Called by the websocket protocol when the connection is established."""
         self.factory.on_client_open(self)
         self.log.info('connection established')
 
     def onClose(self, wasClean, code, reason):
+        """Called by the websocket protocol when the connection is closed."""
         self.factory.on_client_close(self)
         self.log.info('connection closed', reason=reason)
 
     def onMessage(self, payload: Union[bytes, str], isBinary: bool) -> None:
+        """Called by the websocket protocol when a new message is received."""
         self.log.debug('new message', payload=payload.hex() if isinstance(payload, bytes) else payload)
-        self.factory.handle_message(self, payload)
+        if isinstance(payload, bytes):
+            message = json_loadb(payload)
+        else:
+            message = json_loads(payload)
+
+        _type = message.get('type')
+
+        if _type == 'ping':
+            self._handle_ping(message)
+        elif _type == 'subscribe_address':
+            self.factory._handle_subscribe_address(self, message)
+        elif _type == 'unsubscribe_address':
+            self.factory._handle_unsubscribe_address(self, message)
+        elif _type == 'request:history:xpub':
+            self._open_history_xpub_streamer(message)
```

Comment:
Maybe we should disable the streaming by default and require a `--enable-wallet-streaming-api` to enable it.

Reply:
I've added some protection and also some logging so we can block abusers. Do you think it would be enough for making it publicly available now?

Reply:
Fine for me!

Reply:
I agree. I added a `--disable-ws-history-streaming` parameter. What do you think?

Reply:
Thanks! I'm closing this discussion for now. Feel free to reopen if you wish.